documentation/bolt_types_reference.md

```diff
@@ -0,0 +1,319 @@
+# Bolt data types
+
+## `ApplyResult`
+
+An `ApplyResult` is returned from an [apply action](https://puppet.com/docs/bolt/latest/applying_manifest_blocks.html#)
```

Comment:
I think this can be relative, so just `[apply action](applying_manifest_blocks.md#)